lib/proxy/router.go

```diff
@@ -259,6 +271,37 @@ func (r *Router) DialHost(ctx context.Context, clientSrcAddr, clientDstAddr net.
 		case serverAddr == "" && target.GetUseTunnel():
 			serverAddr = reversetunnelclient.LocalNode
 		}
+		// If the node is a registered openssh node don't set agentGetter
+		// so a SSH user agent will not be created when connecting to the remote node.
+		if target.IsOpenSSHNode() {
+			agentGetter = nil
+			isAgentlessNode = true
+
+			// If the node is of SubKindOpenSSHNode, create the signer.
+			//
+			// Creating the ssh.Signer for SubKindOpenSSHEICENode requires the SSH Login User.
```

Comment:
We have effectively `switch target.GetSubKind()` here, but the default branch is empty. If `IsOpenSSHNode()` changes in the future, this might lead to this code behaving unexpectedly. We could guard against it by checking if the `SubKind` has an unexpected value, i.e. not any of those two cases below.

Reply:
I'll create a switch where the `default` branch returns an error